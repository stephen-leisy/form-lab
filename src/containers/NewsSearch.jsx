import React, { Component } from 'react';
import ArticleList from '../components/news/ArticleList';
import { getAllNews, searchNews } from '../services/newsApi';
import Controls from '../components/news/Controls';

export default class NewsSearch extends Component {
  state = {
    articleList: [],
    articleSearch: '',
    loading: true,
  };

  async componentDidMount() {
    const articleList = await getAllNews();
    this.setState({ articleList, loading: false });
  }

  handleSearchChange = ({ target }) => {
    this.setState({ articleSearch: target.value });
  };

  handleSubmit = async (event) => {
    event.preventDefault();
    this.setState({ loading: true });
    const articleList = await searchNews(this.state.articleSearch);
    this.setState({ articleList, loading: false });
  };

  render() {
    const { articleList, loading, articleSearch } = this.state;

    if (loading) return <h2>Loading...</h2>;
    return (
      <>
        <Controls
          search={articleSearch}
          onChange={this.handleSearchChange}
          onSubmit={this.handleSubmit}
        />
        <ArticleList articles={articleList} />
      </>
    );
  }
}
