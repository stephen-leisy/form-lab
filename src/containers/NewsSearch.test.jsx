import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NewsSearch from './NewsSearch';
import dotenv from 'dotenv';
dotenv.config();

describe('test news api endpoint', () => {
  it('returns a list of news stories', async () => {
    render(<NewsSearch />);

    screen.getByText('Loading...');
    const ul = await screen.findByRole('list', { name: 'news-articles' });
    expect(ul).not.toBeEmptyDOMElement();

    const input = await screen.findByLabelText('Search For News');
    userEvent.type(input, 'Portland Trail Blazers pound pacers skid');

    const button = await screen.findByRole('button', { name: 'find-news' });
    userEvent.click(button);

    return waitFor(() => {
      const news = screen.getAllByText('Portland Trail Blazers', {
        exact: false,
      });
      expect(news).toHaveLength(1);
      expect(news).toMatchSnapshot();
    });
  });
});
