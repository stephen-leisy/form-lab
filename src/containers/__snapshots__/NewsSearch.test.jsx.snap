// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test news api endpoint returns a list of news stories 1`] = `
Array [
  <li>
    TITLE: 
    Trail Blazers pound Pacers, snap five-game skid - Reuters
    BY: 
    
    DESCRIPTION: 
    Anfernee Simons' 27 points on 9-of-10 shooting from 3-point range led a dominant effort by the Portland Trail Blazers during a 133-112 win over the Indiana Pacers on Tuesday night in Indianapolis.
  </li>,
]
`;
